### PR TITLE
[Serve][Doc] Add object detection example

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

Add object detection example (from https://github.com/ray-project/ray/pull/33431)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/BUILD [hunk 1]
```diff
@@ -131,6 +131,7 @@ py_test_run_all_subdirectory(
     exclude = [
         "source/serve/doc_code/distilbert.py",
         "source/serve/doc_code/stable_diffusion.py",
+         "source/serve/doc_code/object_dection.py",
```
Comment (richardliaw) [inline doc/BUILD:134]: `detection` is spelled incorrectly (dection)

**Comment by richardliaw** [comment]
1 change (naming typo) but otherwise looks good!

File: doc/source/serve/tutorials/object-detection.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```{literalinclude} ../doc_code/object_detection.py
+:language: python
+:start-after: __example_code_start__
+:end-before: __example_code_end__
+```
+
+Use `serve run object_detection:entrypoint` to start the serve application.
+
+:::{note}
+The autoscaling config sets `min_replicas` to 0, which means the deployment starts with no `ObjectDetection` replicas. These replicas spawn only when a request arrives. After a period where no requests arrive, Serve downscales `ObjectDetection` back to 0 replica to save GPU resources.
```
Comment (edoakes) [inline doc/source/serve/tutorials/object-detection.md:24]:
```suggestion
The autoscaling config sets `min_replicas` to 0, which means the deployment starts with no `ObjectDetection` replicas. These replicas spawn only when a request arrives. After a period where no requests arrive, Serve downscales `ObjectDetection` back to 0 replicas to save GPU resources.
```

**Review by edoakes — APPROVED** [review]

**Comment by sihanwang41** [comment]
^^ @zhe-thoughts doc change

**Review by richardliaw — APPROVED** [review]

**Pull request merged**